Sources/TurnsController.cpp

```diff
@@ -1 +1,51 @@
-﻿#include "../Headers/TurnsController.h"
+﻿#include <algorithm>
+#include <chrono>
+#include <random>
+#include "../Headers/TurnsController.h"
+#include "../Headers/GameConsole.h"
+#include "../Headers/Player.h"
+
+void TurnsController::Initialize(std::vector<std::shared_ptr<Player>> players)
+{
+    players_ = std::move(players);
+    players_amount_ = players_.size();
+}
+
+void TurnsController::SetupTurns()
+{
+    ShufflePlayers();
+    current_player = *players_.begin();
+}
+
+void TurnsController::PlayTurn()
+{
+    current_player->Play();
+}
+
+void TurnsController::NextTurn()
+{
+    NextPlayer();
+}
+
+void TurnsController::NextPlayer()
+{
+    int direction;
+    play_order_ == EPlayOrder::Clockwise ? direction = 1 : direction = -1;
+    current_player_index_ = (current_player_index_ + direction) % players_amount_;
```

Comment:
vc precisa considerar a situação negativa, se for menor que zero, não vai retornar para o último player

Reply:
Verdade. Adicionei um define nessa classe pra o "modular arithmetics" ` ((a % n) + n) % n` baseado nesse video: https://www.youtube.com/watch?v=gy3Ni-wFYR4